rfcs/0000-allow-script-multiple-matches-on-identical-code/0000-allow-script-multiple-matches-on-identical-code.md

```diff
@@ -0,0 +1,96 @@
+---
+Number: "0000"
+Category: Consensus (Hard Fork)
+Status: Draft
+Author: Ian Yang
+Organization: Nervos Foundation
+Created: 2021-02-03
+---
+
+# Allow Multiple Cell Dep Matches When There Is No Ambiguity
+
+## Abstract
+
+This document proposes a transaction verification consensus change to allow multiple cell dep matches on type script hash when all the matches are resolved to the same script code.
+
+## Motivation
+
+Currently, CKB does not allow multiple matches when resolving the script by type script hash in dep cells. There are two scenarios in which this can cause troubles:
+
+1. Dep group is a feature to let transactions add multiple dep cells together. When a user adds multiple dep groups, these groups may have duplicated cells. The user has to expand the dep groups and remove the duplicates, which leads to a larger transaction.
+2. In protocols like Open Transaction, the transaction is assembled by merging several partial transactions. If these partial transactions contain duplicated dep groups, it will make the signature algorithm very complex.
```

Comment:
Currently, [CKB doesn't allow duplicated dep groups](https://github.com/nervosnetwork/ckb/blob/bea2fa3e0fa7cbf33a0f27f31cdaf9df348995ac/verification/src/transaction_verifier.rs#L405-L413). Do you mean "overlapping dep groups"?